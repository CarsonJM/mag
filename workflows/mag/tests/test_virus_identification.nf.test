nextflow_workflow {

    name "Test workflow: MAG"
    script "../main.nf"
    workflow "MAG"
    tag "workflows"
    tag "mag"
    tag "mag_test_virus_identification"

    test("Parameters: run_virus_identification = true") {
        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out
                ).match() }
            )
        }
    }
}
