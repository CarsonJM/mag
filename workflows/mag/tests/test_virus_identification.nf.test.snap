{
    "Parameters: run_virus_identification = true": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "test_minigut_1.merged.fastq.gz:md5,b7eaf06531000c4d32a48d050c2df34f",
                            "test_minigut_2.merged.fastq.gz:md5,a1629ec2f2931892a2a0c43f0f677056"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R1.fastq.gz",
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R2.fastq.gz"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "group-0",
                            "group": "0",
                            "single_end": false,
                            "assembler": "MEGAHIT"
                        },
                        "MEGAHIT-group-0.contigs.fa:md5,8419d4d6c72ac1006324a9fee1394646"
                    ]
                ],
                "2": [
                    
                ],
                "3": [
                    [
                        {
                            "id": "group-0",
                            "group": "0",
                            "single_end": false,
                            "assembler": "MEGAHIT"
                        },
                        "MEGAHIT-group-0.contigs_virus_summary.tsv:md5,b08861c8a938b3eaf8c1d8e533935042"
                    ]
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "7": [
                    
                ],
                "8": [
                    
                ],
                "9": [
                    "versions.yml:md5,3222fd64337a6c070682b63fc5e0cb3c",
                    "versions.yml:md5,85a05e91a1084dbca180ff6ff7e67400",
                    "versions.yml:md5,e2ed4d9a5b5d63ae1a6eef0418d976f0"
                ],
                "assemblies": [
                    [
                        {
                            "id": "group-0",
                            "group": "0",
                            "single_end": false,
                            "assembler": "MEGAHIT"
                        },
                        "MEGAHIT-group-0.contigs.fa:md5,8419d4d6c72ac1006324a9fee1394646"
                    ]
                ],
                "bin_summary": [
                    
                ],
                "genomad": [
                    [
                        {
                            "id": "group-0",
                            "group": "0",
                            "single_end": false,
                            "assembler": "MEGAHIT"
                        },
                        "MEGAHIT-group-0.contigs_virus_summary.tsv:md5,b08861c8a938b3eaf8c1d8e533935042"
                    ]
                ],
                "metaeuk": [
                    
                ],
                "prodigal": [
                    
                ],
                "prokka": [
                    
                ],
                "refined_bins": [
                    
                ],
                "refined_unbins": [
                    
                ],
                "short_reads": [
                    [
                        {
                            "id": "test_minigut",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "test_minigut_1.merged.fastq.gz:md5,b7eaf06531000c4d32a48d050c2df34f",
                            "test_minigut_2.merged.fastq.gz:md5,a1629ec2f2931892a2a0c43f0f677056"
                        ]
                    ],
                    [
                        {
                            "id": "test_minigut_sample2",
                            "group": "0",
                            "single_end": false
                        },
                        [
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R1.fastq.gz",
                            "/nf-core/test-datasets/raw/mag/test_data/test_minigut_sample2_R2.fastq.gz"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3222fd64337a6c070682b63fc5e0cb3c",
                    "versions.yml:md5,85a05e91a1084dbca180ff6ff7e67400",
                    "versions.yml:md5,e2ed4d9a5b5d63ae1a6eef0418d976f0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-09T15:30:48.208307453"
    }
}